sstables/sstables.cc

```diff
@@ -2817,6 +2817,10 @@ thread_local cached_file::metrics index_page_cache_metrics;
 thread_local mc::cached_promoted_index::metrics promoted_index_cache_metrics;
 static thread_local seastar::metrics::metric_groups metrics;
 
+size_t space_used_by_index_cache() {
+    return partition_index_cache::shard_stats().used_bytes + index_page_cache_metrics.cached_bytes;
```

Comment:
Maybe a cleaner way to achieve that would be to have a notion of region zones. When you create caches, instead of passing `get_cache_tracker().region()` you would get `get_cache_tracker().get_index_zone()` which returns 

```
struct region_zone {
    region& r;
    allocator& alloc;
}
```

And alloc is a zone-specific wrapper around region allocator which accounts memory allocated through the zone, like the current `measuring_allocator`.


In caches, instead of:
```
with_allocator(_region.allocator(), [&] {
``` 

you'd do:
```
with_allocator(_zone.allocator(), [&] {
```